src/Downloader.ts

```diff
@@ -184,45 +182,61 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
-    try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
-    } catch (err) {
-      this.mwCapabilities.restApiAvailable = false;
-      logger.warn(`Failed to get remote Rest API`);
+  public async setBaseUrls(){
+    const isVeApiAvailable = this.mwCapabilities.veApiAvailable;
+    const isMobileApiAvailable = this.mwCapabilities.mobileRestApiAvailable;
+    const isDesktopApiAvailable = this.mwCapabilities.desktopRestApiAvailable;
```

Comment:
I don't believe we need this variable, use directly values in `this.mwCapabilities`.

Reply:
I tried that as well, but that is making the code really hard to read and that is why I switched to usage of variables, for clean code and better understandability.